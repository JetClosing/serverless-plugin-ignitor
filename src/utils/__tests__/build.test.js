const path = require('path');
const build = require('../build');
const { read, exists } = require('../file');

const BUILD_DIR = 'ignitor';

const MOCK_FUNCTION_KEY = 'test';

const MOCK_HANDLER = 'src/handler.default';

test('prebuild', () => {
  build.prebuild();
  expect(exists(BUILD_DIR)).toBeTruthy();
});

describe('wrap', () => {
  test('use default wrapper', () => {
    const response = build.wrap(MOCK_FUNCTION_KEY, MOCK_HANDLER, undefined);
    expect(response).toEqual(`${BUILD_DIR}/${MOCK_FUNCTION_KEY}.handler`);

    const contents = read(path.resolve(BUILD_DIR, 'test.js'));
    expect(contents).toMatchSnapshot();
  });

  test('uses override wrapper', () => {
    const wrapperOverride = 'src/wrappers.auth';
    const response = build.wrap(MOCK_FUNCTION_KEY, MOCK_HANDLER, wrapperOverride);
    expect(response).toEqual(`${BUILD_DIR}/${MOCK_FUNCTION_KEY}.handler`);

    const contents = read(path.resolve(BUILD_DIR, 'test.js'));
    expect(contents).toMatchSnapshot();
  });
});

test('clean', () => {
  build.clean();
  expect(exists(BUILD_DIR)).toBeFalsy();
});
