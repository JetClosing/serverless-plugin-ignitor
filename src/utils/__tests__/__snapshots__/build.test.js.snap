// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wrap use default wrapper 1`] = `
"const original = require('../src/handler').default;
const wrapper = require('./ignitor').ignitor;

module.exports = { handler: wrapper(original) };"
`;

exports[`wrap uses override wrapper 1`] = `
"const original = require('../src/handler').default;
const wrapper = require('../src/wrappers').auth;

module.exports = { handler: wrapper(original) };"
`;
